### PR TITLE
Memo config and flag

### PR DESCRIPTION
**Pull request by agouin**
Adds default memo to relayed transactions with `rly` and the version, e.g. `rly(v2.0.0)`.

Memo can be customized through the config file `memo` property, or the `--memo` flag. The flag will override the config property if provided.

Custom memos will have `rly` and the version appended, e.g. `memo` of `Strangelove Ventures` running on relayer version `v2.0.0` would result in a transaction memo of `Strangelove Ventures | rly(v2.0.0)`

### CODE REVIEW

File: cmd/flags.go [hunk 1]
```diff
@@ -303,6 +304,14 @@ func processorFlags(v *viper.Viper, cmd *cobra.Command) *cobra.Command {
 	return cmd
 }
 
+func memoFlag(v *viper.Viper, cmd *cobra.Command) *cobra.Command {
+	cmd.Flags().String(flagMemo, "", "a memo to include in relayed packets")
```
Comment (DavidNix) [inline cmd/flags.go:308]: Do we need highlight we always append a suffix like `| rly(v2.0.0)`? I'd be a little surprised.

**Review by DavidNix — APPROVED** [review]
It's a surprising amount of function signature changes. 

Typically, this indicates a design for a `struct` that can have common fields like `memo` and `log`. And then you write methods off the struct which can use the fields. But that refactor is likely not worth it for this PR.

File: cmd/flags.go (hunk 1 above)
Comment (mark-rushakoff) [reply cmd/flags.go:308]: Let's offer a way to omit the memo. I think empty defaulting to the version is reasonable. "-" as a special case to omit the memo seems reasonably typical.

Comment (agouin) [reply cmd/flags.go:308]: Documented the behavior in the README and added the behavior to omit using `-`

**Pull request merged**